tests/cpp/multidevice.h

```diff
@@ -46,17 +47,29 @@ class MultiDeviceEnvironment : public testing::Environment {
 
 class MultiDeviceTest : public NVFuserTest {
  public:
+  // Given an aten tensor, TensorView the tensor is bound to, and deviceId
+  // returns a shard of the tensor according the sharding annotation in tv
+  // for the deviceId. If tensor is not sharded returns the original tensor.
+  // TODO: If deviceId is not part of the mesh this should return an empty
+  // tensor currently, we don't support this, so for now it returns a slice.
   static at::Tensor shardTensor(
       at::Tensor tensor,
-      const DeviceMesh& mesh,
+      TensorView* tv,
       DeviceIdxType deviceId) {
-    int i = 0;
-    auto devices = mesh.vector();
-    auto it = find(devices.begin(), devices.end(), deviceId);
-    if (it != devices.end()) {
-      i = std::distance(devices.begin(), it);
+    if (isSharded(tv)) {
+      auto sharded_dim = 0;
+      int i = 0;
+      const auto& devices = tv->getDeviceMesh().vector();
+      auto it = std::find(devices.begin(), devices.end(), deviceId);
+      if (it != devices.end()) {
+        i = std::distance(devices.begin(), it);
+      }
+      std::vector<at::indexing::TensorIndex> indices(
+          tensor.dim(), at::indexing::Slice());
+      indices[sharded_dim] = at::indexing::Slice(i, i + 1);
+      return tensor.index(indices).contiguous();
```

Comment:
```suggestion
      return tensor.slice(sharded_dim, i, i + 1).squeeze(sharded_dim).contiguous();
```
and then you don't need `indices`. I think this is simpler code that does the same thing. `at::index` is not very convenient when dealing with slices.